resources/resincli.robot

```diff
@@ -69,37 +71,44 @@ Device "${device_uuid}" log should contain "${value}"
     ${result} =  Run Process    resin device ${device_uuid} | grep OS | rev | cut -d ' ' -f 1 | rev     shell=yes
     Process ${result}
     Should Contain    ${result.stdout}    ${os_version}
 
 Add ENV variable "${variable_name}" with value "${variable_value}" to application "${application_name}"
-    ${result} =  Run Process    resin   env     add     ${variable_name}    ${variable_value}    -a  ${application_name}
-    Process ${result}
-
-Check if ENV variable "${variable_name}" exists in application "${application_name}"
-    ${result_vars} =  Run Process    resin envs -a ${application_name} | sed '/ID[[:space:]]*NAME[[:space:]]*VALUE/,$!d'   shell=yes
-    Process ${result_vars}
-    ${result_name} =  Run Process    echo "${result_vars.stdout}" | grep ${variable_name} | cut -d ' ' -f 2    shell=yes
-    Process ${result_name}
-    Should Contain    ${result_name.stdout}    ${variable_name}
-
-Check if value of ENV variable is "${variable_value}" in application "${application_name}"
-    ${result_vars} =  Run Process    resin envs -a ${application_name} | sed '/ID[[:space:]]*NAME[[:space:]]*VALUE/,$!d'   shell=yes
-    Process ${result_vars}
-    ${result_value} =  Run Process    echo "${result_vars.stdout}" | grep ${variable_value} | cut -d ' ' -f 3    shell=yes
-    Process ${result_value}
-    Should Contain    ${result_value.stdout}    ${variable_value}
-
-Remove ENV variable "${variable_name}" from application "${application_name}"
-    ${result_vars} =  Run Process    resin envs -a ${application_name} | sed '/ID[[:space:]]*NAME[[:space:]]*VALUE/,$!d'   shell=yes
-    Process ${result_vars}
-    ${result_id} =  Run Process    echo "${result_vars.stdout}" | grep ${variable_name} | cut -d ' ' -f 1    shell=yes
+    ${result} =  Run Process    resin env add ${variable_name} ${variable_value} -a ${application_name}    shell=yes
+    Process ${result}
+    
+Check if ENV variable "${variable_name}" with value "${variable_value}" exists in application "${application_name}"
+    ${result_env} =  Run Process    resin envs -a ${application_name} --verbose | sed '/ID[[:space:]]*NAME[[:space:]]*VALUE/,$!d'   shell=yes
+    Process ${result_env}
+    ${result} =  Run Process    echo "${result_env.stdout}" | grep ${variable_name} | grep " ${variable_value}"     shell=yes
+    Process ${result}
+
+Remove ENV variable "${variable_name}" with value "${variable_value}" from application "${application_name}"
+    ${result_env} =  Run Process    resin envs -a ${application_name} --verbose | sed '/ID[[:space:]]*NAME[[:space:]]*VALUE/,$!d'    shell=yes
+    Process ${result_env}
+    ${result_id} =  Run Process    echo "${result_env.stdout}" | grep ${variable_name} | grep " ${variable_value}" | cut -d ' ' -f 1    shell=yes
     Process ${result_id}
     ${result} =  Run Process    resin env rm ${result_id.stdout} --yes     shell=yes
     Process ${result}
 
+Push output console with "${message}"
```

Comment:
This keyword assumes that the git_url push has been done above - In order to keep this independent of that - maybe we can do a fresh clone in a separate directory and git push from there. My rationale is that we should either capture dependencies properly or the keyword should be independent of most previous keywords / states. Thoughts?